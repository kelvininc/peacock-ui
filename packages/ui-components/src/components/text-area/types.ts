import { EventEmitter } from '@stencil/core';
import { EIconName, EOtherIconName } from '../icon/icon.types';

export interface ITextArea {
	/** (optional) Icon to show to the left of the text field */
	icon?: EIconName | EOtherIconName;
	/** (optional) The text to show inside the text area */
	text?: string;
	/** (optional) The placeholder to show in the text area */
	placeholder?: string;
	/** (required) The maximum number of characters allowed */
	maxCharLength: number;
}

export interface ITextAreaEvents {
	/** Emits the current text when there's a change */
	textChange: EventEmitter<string>;
	/** Emits the current text when the text area loses focus */
	textChangeBlur: EventEmitter<string>;
}
