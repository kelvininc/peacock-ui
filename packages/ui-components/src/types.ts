export * from './components';
export * from './utils/types';
export * from './components/action-button/action-button.types';
export * from './components/breadcrumb-item/breadcrumb-item.types';
export * from './components/illustration/illustration.types';
export * from './components/tab-navigation/tab-navigation.types';
export * from './components/text-field/text-field.types';
export * from './components/tooltip/tooltip.types';
export * from './components/summary-card/summary-card.types';
export * from './components/search/search.types';
