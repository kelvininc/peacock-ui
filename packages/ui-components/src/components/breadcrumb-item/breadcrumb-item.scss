@import '../../assets/styles/globals';

.breadcrumb-item {
	@include kv-font-label-small-regular();
	display: inline;
	flex-shrink: 0;
	cursor: pointer;
	user-select: none;
	color: kv-color('neutral-5');

	a {
		color: unset;
		text-decoration: none;

		&:hover {
			text-decoration: underline;
			color: kv-color('primary');
		}
	}

	&.active {
		@include kv-font-label-small-semibold();
		pointer-events: none;
	}
}

.separator {
	@include kv-font-label-small-regular();
	margin: 0 $spacing-2x;
	user-select: none;
	color: kv-color('neutral-5');
}
