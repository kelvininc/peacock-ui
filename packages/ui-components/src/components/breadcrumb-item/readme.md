# *<kv-breadcrumb-item>*



<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- With all properties (only label is mandatory) -->
<kv-breadcrumb-item
	label="Your label here"
	link="Your link here"
	[target]="EBreadcrumbItemTarget.NewTab"
	separator='/'
	active>
</kv-breadcrumb-item>
```


### Javascript

```html
<!-- With all properties (only label is mandatory) -->
<kv-breadcrumb-item
	label="Your label here"
	link="Your link here"
	target="_blank"
	separator='/'
	active>
</kv-breadcrumb-item>
```


### React

```tsx
import React from 'react';
import { KvBreadcrumbItem } from '@kelvininc/react-ui-components';
export const KvBreadcrumbItemExample: React.FC = () => (
  <>
    {/*-- With all properties (only label is mandatory) --*/}
	<KvBreadcrumbItem
		label="Your label here"
		link="Your link here"
		target={EBreadcrumbItemTarget.NewTab}
		separator='/'
		active>
	</KvBreadcrumbItem>
  </>
);
```


### Stencil

```tsx
import { Component, h } from '@stencil/core';
@Component({
  tag: 'kv-link-example',
  styleUrl: 'kv-link-example.css',
  shadow: true,
})
export class KvBreadcrumbItemExample {
  render() {
    return (
		<KvBreadcrumbItem
			label="Your label here"
			link="Your link here"
			target={EBreadcrumbItemTarget.NewTab}
			separator='/'
			active>
		</KvBreadcrumbItem>
	);
  }
}
```



## Properties

| Property             | Attribute   | Description                                                                         | Type                                                                                                | Default     |
| -------------------- | ----------- | ----------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------- | ----------- |
| `active`             | `active`    | (optional) Sets this breadcrumb styling to be the active one (usually the last one) | `boolean`                                                                                           | `undefined` |
| `href`               | `href`      | (optional) The breadcrumb's link                                                    | `string`                                                                                            | `undefined` |
| `label` _(required)_ | `label`     | (required) The text to display on the breadcrumb                                    | `string`                                                                                            | `undefined` |
| `separator`          | `separator` | (optional) The separator to use                                                     | `string`                                                                                            | `undefined` |
| `target`             | `target`    | (optional) The target of the link (only used if href is provided)                   | `EAnchorTarget.BrowserDefault \| EAnchorTarget.NewTab \| EAnchorTarget.Parent \| EAnchorTarget.Top` | `undefined` |


## Events

| Event                 | Description                                    | Type                           |
| --------------------- | ---------------------------------------------- | ------------------------------ |
| `breadcrumbItemClick` | Emitted when the user clicks on the breadcrumb | `CustomEvent<IBreadcrumbItem>` |


## Dependencies

### Used by

 - [kv-breadcrumb](../breadcrumb)

### Graph
```mermaid
graph TD;
  kv-breadcrumb --> kv-breadcrumb-item
  style kv-breadcrumb-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


