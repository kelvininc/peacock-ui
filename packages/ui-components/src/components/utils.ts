import { isNil } from 'lodash-es';
import { DEFAULT_CONFIG } from '../globals/config';
import { UIComponentsConfig } from '../types';

export const getConfig = (): UIComponentsConfig => {
	if (typeof window !== 'undefined') {
		const KvUiComponents = window.KvUiComponents;
		if (!isNil(KvUiComponents) && !isNil(KvUiComponents.config)) {
			return KvUiComponents.config;
		}
	}

	return DEFAULT_CONFIG;
};
