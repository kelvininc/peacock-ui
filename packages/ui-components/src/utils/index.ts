export * from './types';
export * as arraysHelper from './arrays.helper';
export * as clipboardHelper from './clipboard.helper';
export * as cssClassHelper from './css-class.helper';
export * as dateHelper from './date.helper';
export * as searchHelper from './search.helper';
export * as stringHelper from './string.helper';
export * as floatingUIHelper from './floating-ui.helper';
export * as mouseEventHelper from './mouse-event.helper';
