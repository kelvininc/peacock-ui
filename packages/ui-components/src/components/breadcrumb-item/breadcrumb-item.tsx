import { Component, Host, h, Prop, EventEmitter, Event } from '@stencil/core';
import { throttle } from 'lodash-es';
import { EAnchorTarget } from '../../utils/types';
import { IBreadcrumbItem, IBreadcrumbItemEvents } from './breadcrumb-item.types';

/**
 * @part anchor - The anchor element.
 */
@Component({
	tag: 'kv-breadcrumb-item',
	styleUrl: 'breadcrumb-item.scss',
	shadow: true
})
export class KvBreadcrumbItem implements IBreadcrumbItem, IBreadcrumbItemEvents {
	/** @inheritdoc */
	@Prop({ reflect: true }) href?: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) target?: EAnchorTarget;
	/** @inheritdoc */
	@Prop({ reflect: true }) download?: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) label!: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) active?: boolean;

	/** @inheritdoc */
	@Event() breadcrumbItemClick: EventEmitter<IBreadcrumbItem>;

	connectedCallback() {
		this.clickThrottler = throttle(() => this.onItemClick(), 300);
	}

	private clickThrottler: () => void;
	private onItemClick = () => {
		this.breadcrumbItemClick.emit({
			label: this.label,
			href: this.href,
			target: this.target,
			active: this.active
		});
	};

	render() {
		return (
			<Host>
				<div
					class={{
						'breadcrumb-item': true,
						'breadcrumb-item--active': this.active
					}}
				>
					<a href={this.href} target={this.target} onClick={this.clickThrottler} part="anchor">
						{this.label}
					</a>
				</div>
			</Host>
		);
	}
}
