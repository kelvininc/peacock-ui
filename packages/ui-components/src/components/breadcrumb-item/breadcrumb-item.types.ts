import { EventEmitter } from '@stencil/core';
import { JSX } from '@stencil/core/internal';
import { IAnchor } from '../../utils/types';

export interface IBreadcrumbItem extends IAnchor {
	/** (required) The text to display on the breadcrumb */
	label: string;
	/** (optional) Sets this breadcrumb styling to be the active one (usually the last one) */
	active?: boolean;
	/** (optional) Adds a custom breadcrumb label */
	customLabel?: JSX.Element;
}

export interface IBreadcrumbItemEvents {
	/** Emitted when the user clicks on the breadcrumb */
	breadcrumbItemClick: EventEmitter<IBreadcrumbItem>;
}
