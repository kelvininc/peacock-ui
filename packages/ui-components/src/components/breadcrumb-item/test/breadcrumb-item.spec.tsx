import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { KvBreadcrumbItem } from '../breadcrumb-item';
import { EAnchorTarget } from '../../../utils/types';
import { h } from '@stencil/core'

describe('KvBreadcrumbItem (unit tests)', () => {
	let page: SpecPage;
	let component: KvBreadcrumbItem;

	beforeEach(async () => {
		page = await newSpecPage({
			components: [KvBreadcrumbItem],
			template: () => (
				<kv-breadcrumb-item
					label='Awesome Label'
					href='https://kelvin.ai'
					target={EAnchorTarget.NewTab}
					active>
				</kv-breadcrumb-item>
			),
		});
		component = page.rootInstance;
	});

	describe('when the component loads', () => {
		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should set the correct label', () => {
			expect(component.label).toEqual('Awesome Label');
		});

		it('should set the correct link', () => {
			expect(component.href).toEqual('https://kelvin.ai');
		});

		it('should set the correct target', () => {
			expect(component.target).toEqual(EAnchorTarget.NewTab);
		});

		it('should set the correct active status', () => {
			expect(component.active).toEqual(true);
		});
	});
});
