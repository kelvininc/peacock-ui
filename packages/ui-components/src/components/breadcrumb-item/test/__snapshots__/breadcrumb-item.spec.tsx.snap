// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KvBreadcrumbItem (unit tests) when the component loads should match the snapshot 1`] = `
<kv-breadcrumb-item active="" label="Awesome Label">
  <mock:shadow-root>
    <div class="breadcrumb-item breadcrumb-item--active">
      <slot>
        Awesome Label
      </slot>
    </div>
  </mock:shadow-root>
</kv-breadcrumb-item>
`;
